### PR TITLE
fix: Fix error types

### PR DESCRIPTION
**Pull request by Shehryar21**
## Implementation description
* Even though `docker compose up` did not give me issues, but on vscode all the errors were underlined in red and throwing errors. This pr fixes that. I used https://github.com/uwblueprint/ccbc/pull/17 as a guide.

https://devblogs.microsoft.com/typescript/announcing-typescript-4-0/#unknown-on-catch. This also explains more on why we need unknown for error type

## Steps to test
1. Make sure I didnt break anything, so run `docker compose up` and then test some endpoints to make sure everything is working fine.

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/typescript/middlewares/validators/authValidators.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ export const registerRequestValidator = async (
   if (!validatePrimitive(req.body.password, "string")) {
     return res.status(400).send(getApiValidationError("password", "string"));
   }
-  if (!validatePrimitive(req.body.password, "string")) {
```
Comment (annechung) [inline backend/typescript/middlewares/validators/authValidators.ts:43]: should we change this back

File: backend/typescript/middlewares/validators/authValidators.ts [hunk 2]
```diff
@@ -40,7 +40,7 @@ export const registerRequestValidator = async (
...
+  if (!validatePrimitive(req.body.phoneNumber, "string")) {
```
Comment (annechung) [inline backend/typescript/middlewares/validators/authValidators.ts:43]: should we change this back?

File: backend/typescript/utilities/errorMessageUtil.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+const getErrorMessage = (error: unknown): string => {
+  return error instanceof Error
+    ? getErrorMessage(error)
```
Comment (annechung) [inline backend/typescript/utilities/errorMessageUtil.ts:3]: doesn't this cause an infinite loop :o 
```suggestion
    ? error.message
```

Comment (Shehryar21) [reply backend/typescript/utilities/errorMessageUtil.ts:3]: lol yes thats why you dont do replace all

File: backend/typescript/middlewares/validators/authValidators.ts (hunk 2 above)
Comment (Shehryar21) [reply backend/typescript/middlewares/validators/authValidators.ts:43]: so this one was supposed to be phoneNumber, the if statement before checks for password

Comment (annechung) [reply backend/typescript/middlewares/validators/authValidators.ts:43]: ohhh right ok

**Review by annechung — APPROVED** [review]

**Pull request merged**